UnityProject/Assets/Scripts/PlayGroups/PlayerMove.cs

```diff
@@ -193,69 +193,79 @@ private Vector3Int GetMoveDirection(KeyCode action)
 		/// <summary>
 		///     Check current and next tiles to determine their status and if movement is allowed
 		/// </summary>
-		private Vector3Int AdjustDirection(Vector3Int currentPosition, Vector3Int direction)
+		private Vector3Int AdjustDirection(Vector3Int currentPosition, Vector3Int direction, bool isReplay)
 		{
 			if (isGhost) {
 				return direction;
 			}
 
-			//Is the current tile restrictive?
 			Vector3Int newPos = currentPosition + direction;
-			Vector3 newRayPos = transform.position + direction;
-			rayHit = Physics2D.RaycastAll(newRayPos, (Vector3)direction, 0.2f, hitCheckLayers);
-			Debug.DrawLine(newRayPos, newRayPos + ((Vector3)direction * 0.2f), Color.red, 1f);
-			for (int i = 0; i < rayHit.Length; i++) {
-				//checks to see if the matrix has changed
-				if (rayHit[i].collider.gameObject.layer == 24
-				   && rayHit[i].collider != curMatrixCol) {
-					curMatrixCol = rayHit[i].collider;
-					ChangeMatricies(rayHit[i].collider.gameObject.transform.parent);
-					Debug.Log("Change Matricies");
-				}
 
-				//Detected windows or walls (global will detect on other matricies also)
-				if (rayHit[i].collider.gameObject.layer == 9
-				   || rayHit[i].collider.gameObject.layer == 18) {
-					return Vector3Int.zero;
-				}
+			//isReplay tells AdjustDirection if the move being carried out is a replay move for prediction or not
+			//a replay move is a move that has already been carried out on the LocalPlayer's client
+			if (!isReplay) {
+				
+				Vector3 newRayPos = transform.position + direction;
+				rayHit = Physics2D.RaycastAll(newRayPos, (Vector3)direction, 0.2f, hitCheckLayers);
+			//	Debug.DrawLine(newRayPos, newRayPos + ((Vector3)direction * 0.2f), Color.red, 1f);
+
+				//Detect new matrices
+				for (int i = 0; i < rayHit.Length; i++) {
+					//checks to see if the matrix has changed
+					if (rayHit[i].collider.gameObject.layer == 24
```

Comment:
The reason I haven't yet is because the number and its layer name can be found so easy by just hitting the layer drop down on any object. But we can have a public enum to make it easier in the future